internal/apiclient/apiclient.go

```diff
@@ -224,8 +224,8 @@ func NewKopiaAPIClient(options Options) (*KopiaAPIClient, error) {
 		tp, _ := transport.(*http.Transport)
 		transport = tp.Clone()
 		tp, _ = transport.(*http.Transport)
-		tp.DialContext = func(_ context.Context, _, _ string) (net.Conn, error) {
-			dial, err := net.Dial("unix", u.Path)
+		tp.DialContext = func(ctx context.Context, _, _ string) (net.Conn, error) {
```

Comment:
It is not completely clear (to me) what the side effect of the `ctx` changes might be w.r.t context cancellation.

Can you put the `ctx` changes in a separate PR please? maybe by disabling the `noctx` linter in the config for now.

That way, all the other changes in this PR, including the linter upgrade. Then the `ctx` changes can be merged after the release.

Reply:
- #4972